packages/cfd/src/Containers/__tests__/dmt5-trade-modal.spec.tsx

```diff
@@ -0,0 +1,123 @@
+    const mock_dxtrade_tokens = {
+        demo: '',
+        real: '',
+    };
+
+    afterEach(() => {
+        jest.clearAllMocks();
+    });
+
+    it('should render DMT5TradeModal', () => {
+        render(
+            <DMT5TradeModal
+                mt5_trade_account={mockMt5TradeAccount}
+                show_eu_related_content={false}
+                onPasswordManager={mockOnPasswordManager}
+                toggleModal={mockToggleModal}
+                dxtrade_tokens={mock_dxtrade_tokens}
+            />
+        );
+
+        expect(screen.getByText('Financial')).toBeInTheDocument();
+        expect(screen.getByText(/123.45/)).toBeInTheDocument();
+        expect(screen.getByText(/USD/i)).toBeInTheDocument();
+    });
+
+    it('should show the correct title for non-EU accounts when market_type is synthetic', () => {
+        render(
+            <DMT5TradeModal
+                mt5_trade_account={mockMt5TradeAccount}
+                show_eu_related_content={false}
+                onPasswordManager={mockOnPasswordManager}
+                toggleModal={mockToggleModal}
+                dxtrade_tokens={mock_dxtrade_tokens}
+            />
+        );
+        expect(screen.getByText('Financial')).toBeInTheDocument();
+    });
+
+    it('should show the correct title for non-EU accounts when market_type is synthetic', () => {
+        mockMt5TradeAccount.market_type = 'synthetic';
+        render(
+            <DMT5TradeModal
+                mt5_trade_account={mockMt5TradeAccount}
+                show_eu_related_content={false}
+                onPasswordManager={mockOnPasswordManager}
+                toggleModal={mockToggleModal}
+                dxtrade_tokens={mock_dxtrade_tokens}
+            />
+        );
+        expect(screen.getByText('Derived')).toBeInTheDocument();
+    });
+
+    it('should show the correct title for EU accounts', () => {
+        render(
+            <DMT5TradeModal
+                mt5_trade_account={mockMt5TradeAccount}
+                show_eu_related_content={true}
+                onPasswordManager={mockOnPasswordManager}
+                toggleModal={mockToggleModal}
+                dxtrade_tokens={mock_dxtrade_tokens}
+            />
+        );
+        expect(screen.getByText('CFDs')).toBeInTheDocument();
+    });
+
+    it('should call onPasswordManager when the password edit button is clicked and close it with Toggle Modal', () => {
+        render(
+            <DMT5TradeModal
+                mt5_trade_account={mockMt5TradeAccount}
+                show_eu_related_content={false}
+                onPasswordManager={mockOnPasswordManager}
+                toggleModal={mockToggleModal}
+                dxtrade_tokens={mock_dxtrade_tokens}
+            />
+        );
+        const passwordEditButton = screen.getByTestId('dt_cfd-trade-modal__password-action');
+        fireEvent.click(passwordEditButton);
+        expect(mockOnPasswordManager).toHaveBeenCalledTimes(1);
```

Comment:
onClose is called after the PasswordManager is called.
So both scenarios are tested in the same test case.